presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -243,6 +250,17 @@ private SqlQueryExecution(
                     }
                 }
             }
+
+            // Optionally build and validate plan immediately, before execution begins
+            if (isEagerPlanValidationEnabled(getSession())) {
+                if (eagerPlanValidationExecutorInstance.get() == null) {
+                    eagerPlanValidationExecutorInstance.set(newFixedThreadPool(getEagerPlanValidationThreadPoolSize(getSession()), threadsNamed("presto-eager-plan-validation-%s")));
```

Comment:
I know this probably isn't the right way to setup the threadpool, but not sure the best way to create one where the size is taken from session property. Also, it should probably be conditional on `eager-plan-validation-enabled` being set. Any suggestions?

Reply:
Let's not configure it from the session property at all.  We can always create the threadpool required for this feature.  For deployments that don't use it, to eliminate any overhead, we can just configure the threadpool to start at 0 (the core pool size).    We can configure the max pool size.

Reply:
Gotcha, I'm working on fixing it up